query/transformation.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package query contains all the logic to parse
+// and execute queries against the underlying metric system.
+package query
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/square/metrics/api"
+)
+
+// A transform takes the list of values, other parameters, and the timerange of the query.
+type transform func([]float64, []value, api.Timerange) ([]float64, error)
+
+// timerangeScale takes an api.Timerange and computes the "scale" or duration of one sample (in seconds).
+// It is useful for transformations that normalize on time (like derivative or integral).
+func timerangeScale(timerange api.Timerange) float64 {
```

Comment:
`transform`s now only take the resolution (although I called it `scale`... I think it will be in units of seconds even if resolution is in finer units)